.github/workflows/sync-main-to-rhoai.yaml

```
@@ -10,23 +10,34 @@ jobs:
   sync_main_into_rhoai:
     runs-on: ubuntu-latest
     steps:
+      - name: Generate GitHub App Token
+        id: generate-token
+        uses: actions/create-github-app-token@v2
+        with:
+          app-id: ${{ secrets.ODH_RELEASE_BOT_APP_ID }}
+          private-key: ${{ secrets.ODH_RELEASE_BOT_PRIVATE_KEY }}
+
       - name: Checkout repository
         uses: actions/checkout@v5
         with:
           fetch-depth: 0
           ref: rhoai
+          token: ${{ steps.generate-token.outputs.token }}
```

It's ok to use the default token to checkout the repository, otherwise we also need to set `persist-credentials: false` in actions/checkout.

```suggestion
      - name: Checkout repository
        uses: actions/checkout@v5
        with:
          fetch-depth: 0
          ref: rhoai

      - name: Generate GitHub App Token
        id: generate-token
        uses: actions/create-github-app-token@v2
        with:
          app-id: ${{ secrets.ODH_RELEASE_BOT_APP_ID }}
          private-key: ${{ secrets.ODH_RELEASE_BOT_PRIVATE_KEY }}
```